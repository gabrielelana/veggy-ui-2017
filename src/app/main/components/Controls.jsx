const Controls = ({startDisabled, onStart}) => (
  <div className="column">
    <div className="box">
        <div className="block">
          <button id="startButton" disabled={startDisabled} onClick={onStart} className="button is-primary is-large">Start</button>
        </div>
    </div>
  </div>
)

Controls.defaultProps = {
  startDisabled: false,
  onStart: () => {},
}

export default Controls