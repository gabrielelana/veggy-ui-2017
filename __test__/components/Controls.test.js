import React from 'react'
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme'
import Controls from '../../src/app/main/Controls'

describe('<Controls />', () => {
  
  test('with default props', () => {
    const component = renderer.create(<Controls />)
    expect(component.toJSON()).toMatchSnapshot()
  })

})