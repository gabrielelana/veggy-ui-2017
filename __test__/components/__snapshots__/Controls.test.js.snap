// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls /> with default props 1`] = `
<div
  className="column"
>
  <div
    className="box"
  >
    <div
      className="block"
    >
      <button
        className="button is-primary is-large"
        disabled={false}
        id="startButton"
        onClick={[Function]}
      >
        Start
      </button>
      <button
        className="button is-danger is-large"
        disabled={true}
        id="squashButton"
        onClick={[Function]}
        style={
          Object {
            "marginLeft": "10px",
          }
        }
      >
        Squash
      </button>
    </div>
  </div>
</div>
`;

exports[`<Controls /> with squashDisabled 1`] = `
<div
  className="column"
>
  <div
    className="box"
  >
    <div
      className="block"
    >
      <button
        className="button is-primary is-large"
        disabled={false}
        id="startButton"
        onClick={[Function]}
      >
        Start
      </button>
      <button
        className="button is-danger is-large"
        disabled={true}
        id="squashButton"
        onClick={[Function]}
        style={
          Object {
            "marginLeft": "10px",
          }
        }
      >
        Squash
      </button>
    </div>
  </div>
</div>
`;

exports[`<Controls /> with startDisabled 1`] = `
<div
  className="column"
>
  <div
    className="box"
  >
    <div
      className="block"
    >
      <button
        className="button is-primary is-large"
        disabled={true}
        id="startButton"
        onClick={[Function]}
      >
        Start
      </button>
      <button
        className="button is-danger is-large"
        disabled={true}
        id="squashButton"
        onClick={[Function]}
        style={
          Object {
            "marginLeft": "10px",
          }
        }
      >
        Squash
      </button>
    </div>
  </div>
</div>
`;
